server/src/main/java/io/spine/server/delivery/PickUpOutcome.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2023, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.delivery;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import io.spine.server.NodeId;
+
+import javax.annotation.Nullable;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * An outcome of the {@link ShardedWorkRegistry#pickUp(ShardIndex, NodeId) pickUp()} method of
+ * {@link ShardedWorkRegistry} indicating the result of the pickup attempt.
+ *
+ * <p>The outcome contains one of 3 following results:
+ * <p>1. Shard picked up successfully. In this case the outcome will contain
+ * a new {@link ShardProcessingSession}.
+ * <p>2. Shard could not be picked up as it's already picked by another worker. In this case
+ * the outcome will contain the {@code WorkerId} of the worker owning the session.
+ * <p>3. Shard could not be picked up for some technical problem (Exception). In this case
+ * the outcome will contain the occurred {@code Exception}.
+ */
+public final class PickUpOutcome {
```

Comment:
I think this should be a Proto message, possibly with an interface providing all the convenience methods. 

Otherwise it would be difficult to pass this data over the wire from Liquor.